registry/storage/driver/gcs/gcs.go

```diff
@@ -322,77 +329,50 @@ func (d *driver) Reader(ctx context.Context, path string, offset int64) (io.Read
 		}
 		return nil, err
 	}
-	if res.Header.Get("Content-Type") == uploadSessionContentType {
-		defer res.Body.Close()
+	if r.Attrs.ContentType == uploadSessionContentType {
+		defer r.Close()
 		return nil, storagedriver.PathNotFoundError{Path: path}
 	}
-	return res.Body, nil
-}
-
-func getObject(client *http.Client, bucket string, name string, offset int64) (*http.Response, error) {
-	// copied from cloud.google.com/go/storage#NewReader :
-	// to set the additional "Range" header
-	u := &url.URL{
-		Scheme: "https",
-		Host:   "storage.googleapis.com",
-		Path:   fmt.Sprintf("/%s/%s", bucket, name),
-	}
-	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
-	if err != nil {
-		return nil, err
-	}
-	if offset > 0 {
-		req.Header.Set("Range", fmt.Sprintf("bytes=%v-", offset))
-	}
-	var res *http.Response
-	err = retry(func() error {
-		var err error
-		res, err = client.Do(req)
-		return err
-	})
-	if err != nil {
-		return nil, err
-	}
-	return res, googleapi.CheckMediaResponse(res)
+	return r, nil
 }
 
 // Writer returns a FileWriter which will store the content written to it
 // at the location designated by "path" after the call to Commit.
-func (d *driver) Writer(ctx context.Context, path string, append bool) (storagedriver.FileWriter, error) {
-	writer := &writer{
-		client: d.client,
-		bucket: d.bucket,
-		name:   d.pathToKey(path),
+func (d *driver) Writer(ctx context.Context, path string, appendParam bool) (storagedriver.FileWriter, error) {
```

Comment:
we did that in the S3 driver because `append` is a `Go` reserved word and was triggering people because their editors were highlighting it as a reserved word...

Reply:
To be pedantic, `append` is a built-in function. If it was a reserved word, it'd be a compile error to use it as a variable name. That being said, avoiding the shadowing of a built-in function is a legitimate reason to rename which I'd overlooked before. It's times like this that I wish the Go community had adopted the practice of appending an underscore to the name to resolve these kinds of name collisions, like in the Python community.

A couple of bikeshed ideas for naming the parameter:
- `appendMode`
- `appendTo`

Reply:
> To be pedantic, append is a built-in function. If it was a reserved word, it'd be a compile error to use it as a variable name

Fair comment.

`appendMode` works for me. WDYT @Jamstah 

We should change S3 driver (and possibly Azure one as well) accordingly.

Reply:
Changed to `appendMode`: https://github.com/distribution/distribution/pull/4120/commits/dcaa713cfbee7ca407b72befdbf30f61b347c682